packages/slice-machine/lib/models/common/http/DefaultClient.ts

```diff
@@ -42,26 +47,28 @@ function createApiUrl(base: string, { STAGE, PROD }: ApiSettings): string {
   return PROD;
 }
 
+type Body = Slice<AsObject> | Record<string, unknown> | string;
+
 function createFetcher(
   apiUrl: string,
   repo: string,
   auth: string
 ): (
   prefix: string,
-  body?: object | string,
+  body?: Body,
   action?: string,
   method?: string
 ) => Promise<Response> {
   return function runFetch(
     prefix: string,
-    body?: object | string,
+    body?: Body,
     action = "",
     method = "get"
   ): Promise<Response> {
     const headers = {
       repository: repo,
       Authorization: `Bearer ${auth}`,
-      UserAgent: "slice-machine",
+      ...(typeof window !== undefined ? { "User-Agent": "slice-machine" } : {}),
```

Comment:
the key is much better now 👍.
If I'm not mistaken, this code will be executed only on the server, not in the browser.
So this condition will always be false isn't it ?

Why do we need to differentiate cases, in which case do we want to send the information and in which case do we not ?

Reply:
this is called solely server-side

Reply:
thanks hugo, then I guess we don't need to check if the window exists.

Reply:
done